ts/src/p2b.ts

```diff
@@ -0,0 +1,1263 @@
+        //
+        //    {
+        //        "success": true,
+        //        "errorCode": "",
+        //        "message": "",
+        //        "result": {
+        //            "offset": 0,
+        //            "limit": 50,
+        //            "records": [
+        //                {
+        //                    "id": 7429883128,             // Deal id
+        //                    "time": 1698237535.41196,     // Deal execution time
+        //                    "fee": "0.01755848704",       // Deal fee
+        //                    "price": "34293.92",          // Deal price
+        //                    "amount": "0.00032",          // Deal amount
+        //                    "dealOrderId": 171366551416,  // Deal order id
+        //                    "role": 1,                    // Deal role (1 - maker, 2 - taker)
+        //                    "deal": "10.9740544"          // Total (price * amount)
+        //                }
+        //            ]
+        //        }
+        //    }
+        //
+        const result = this.safeValue (response, 'result', {});
+        const records = this.safeValue (result, 'records', []);
+        return this.parseTrades (records, market, since, limit);
+    }
+
+    async fetchMyTrades (symbol: string = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name p2b#fetchMyTrades
+         * @description fetch all trades made by the user, only the transaction records in the past 3 month can be queried, the time between since and params["until"] cannot be longer than 24 hours
+         * @see https://github.com/P2B-team/p2b-api-docs/blob/master/api-doc.md#deals-history-by-market
+         * @param {string} symbol unified market symbol of the market orders were made in
+         * @param {int} since the earliest time in ms to fetch orders for
+         * @param {int} [limit] 1-100, default=50
+         * @param {object} [params] extra parameters specific to the p2b api endpoint
+         * @param {int} params.until the latest time in ms to fetch orders for
+         *
+         * EXCHANGE SPECIFIC PARAMETERS
+         * @param {int} [params.offset] 0-10000, default=0
+         * @returns {Trade[]} a list of [trade structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#public-trades}
+         */
+        await this.loadMarkets ();
+        const until = this.safeInteger (params, 'until');
+        params = this.omit (params, 'until');
+        if (symbol === undefined) {
+            throw new BadRequest (this.id + ' fetchClosedOrders must have a symbol argument');
+        }
+        if (since === undefined) {
+            throw new BadRequest (this.id + ' fetchClosedOrders must have a since argument');
+        }
+        if (until === undefined) {
+            throw new BadRequest (this.id + ' fetchClosedOrders must have an extra argument params["until"]');
```

Comment:
we should try to have default values to avoid requiring them explicitly

Reply:
besides that the error message is wrong, it refers ftchClosedOrders

Reply:
should 

> we should try to have default values to avoid requiring them explicitly

should we? the default values would need to be this.milliseconds () and this.milliseconds - 24 hours, but this is a behaviour that doesn't match any other exchanges and is way easier to miss if we're providing default values. By throwing an exception, at least the user is aware of whats going on

Reply:
I set defaults anyway cause I figured you're stance wouldn't change, but I thought it was worth bringing up

Reply:
@samgermain I understand, but the goal of ccxt is to unify behaviors across the different exchanges, if usually you don't need to provide any since/until to call this method on other exchanges, here should be the same otherwise the user can't call it in a generic way.